# Project Investigate a Dataset

I choose baseball dataset.

## Phase Simple EDA

Before Before defining the question, I did a simple data search and understood the data set.

In [25]:
# import Library
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [51]:
# load csv(datasets)

# teams = pd.read_csv("/Users/Akihiro/Downloads/baseballdatabank-2017.1/core/Teams.csv")
salaries = pd.read_csv("/Users/Akihiro/Downloads/baseballdatabank-2017.1/core/Salaries.csv")
batting = pd.read_csv("/Users/Akihiro/Downloads/baseballdatabank-2017.1/core/Batting.csv")
pitching = pd.read_csv("/Users/Akihiro/Downloads/baseballdatabank-2017.1/core/Pitching.csv")
players = pd.read_csv("/Users/Akihiro/Downloads/baseballdatabank-2017.1/core/Master.csv")


In [52]:
players.sample(10)

playerID  birthYear  birthMonth  birthDay birthCountry birthState  \
7306   hawpebr01     1979.0         6.0      22.0          USA         TX   
307    anderal01     1914.0         1.0      28.0          USA         GA   
7714    hillke01     1965.0        12.0      14.0          USA         MA   
12722  obriepe04     1990.0         7.0      15.0          USA         FL   
3440   cooneph01     1882.0         9.0      14.0          USA         NY   
14331  rextewi01     1850.0         NaN       NaN          USA         NY   
13309  pegueja01     1981.0         1.0       4.0         D.R.       Azua   
7289   hausmto01     1953.0         3.0      31.0          USA         SD   
2795   caulfja01     1917.0        11.0      23.0          USA         CA   
6833    guthch01     1856.0         NaN       NaN          USA         IL   

         birthCity  deathYear  deathMonth  deathDay    ...       nameLast  \
7306    Fort Worth        NaN         NaN       NaN    ...          Hawpe   
307    Gainesville     1985.0         6.0      23.0    ...       Anderson   
7714          Lynn        NaN         NaN       NaN    ...           Hill   
12722      Hialeah        NaN         NaN       NaN    ...        O'Brien   
3440      New York     1957.0        10.0       6.0    ...         Cooney   
14331     Brooklyn     1898.0         6.0      23.0    ...         Rexter   
13309         Azua        NaN         NaN       NaN    ...        Peguero   
7289      Mobridge        NaN         NaN       NaN    ...        Hausman   
2795   Los Angeles     1986.0        12.0      16.0    ...      Caulfield   
6833       Chicago     1883.0         7.0       5.0    ...           Guth   

             nameGiven weight height bats throws       debut   finalGame  \
7306     Bradley Bonte  190.0   74.0    L      L  2004-05-01  2013-07-22   
307      Alfred Walton  165.0   71.0    R      R  1941-04-20  1946-04-21   
7714      Kenneth Wade  200.0   76.0    R      R  1988-09-03  2001-04-18   
12722     Peter Robert  235.0   76.0    R      R  2015-09-11  2016-09-29   
3440   Philip Clarence  155.0   68.0    L      R  1905-09-27  1905-09-27   
14331       William H.    NaN    NaN  NaN    NaN  1875-09-25  1875-09-25   
13309  Jailen Alphonso  185.0   72.0    R      R  2007-06-08  2008-09-25   
7289    Thomas Matthew  190.0   76.0    R      R  1975-04-26  1982-09-30   
2795       John Joseph  170.0   71.0    R      R  1946-04-24  1946-08-25   
6833        Charles J.    NaN    NaN  NaN    NaN  1880-09-30  1880-09-30   

        retroID    bbrefID  
7306   hawpb001  hawpebr01  
307    andea102  anderal01  
7714   hillk001   hillke01  
12722  obrip002  obriepe01  
3440   coonp101  cooneph01  
14331  rextw101  rextewi01  
13309  peguj002  pegueja01  
7289   haust101  hausmto01  
2795   caulj101  caulfja01  
6833   guthc101   guthch01  

[10 rows x 24 columns]

In [27]:
salaries.sample(5)

yearID teamID lgID   playerID    salary
23486    2013    LAN   NL  billich01  11000000
2572     1988    TOR   AL  musseje01    127000
18418    2007    FLO   NL  treanma01    380000
18123    2007    ARI   NL  hairssc01    389000
10517    1998    CIN   NL  tomkobr01    230000

In [28]:
batting.sample(5)

playerID  yearID  stint teamID lgID    G   AB   R   H  2B  ...    RBI  \
46390  shopato01    1967      1    NYA   AL    8   27   2   8   1  ...    6.0   
83334  asencmi01    2003      1    KCA   AL    8    0   0   0   0  ...    0.0   
35     ewellge01    1871      1    CL1  NaN    1    3   0   0   0  ...    0.0   
57193  simpsjo01    1979      1    SEA   AL  120  265  29  75  11  ...   27.0   
16279  kavanma01    1916      2    CLE   AL   19   44   4  11   2  ...   10.0   

        SB   CS  BB    SO  IBB  HBP   SH   SF  GIDP  
46390  2.0  0.0   1   5.0  1.0  0.0  0.0  1.0   0.0  
83334  0.0  0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  
35     0.0  0.0   0   0.0  NaN  NaN  NaN  NaN   NaN  
57193  6.0  3.0  11  21.0  1.0  1.0  5.0  2.0   5.0  
16279  0.0  NaN   2   5.0  NaN  0.0  0.0  NaN   NaN  

[5 rows x 22 columns]

In [29]:
pitching.sample(5)

playerID  yearID  stint teamID lgID   W   L   G  GS  CG  ...    IBB  \
35197  leiteal01    2003      1    NYN   NL  15   9  30  30   1  ...   11.0   
10173  welchjo01    1936      1    BOS   AL   2   1   9   3   1  ...    NaN   
35762  figuene01    2004      1    PIT   NL   0   3  10   3   0  ...    1.0   
14060  stobbch01    1952      1    CHA   AL   7  12  38  17   2  ...    NaN   
16592  giggibo01    1962      1    KC1   AL   1   1   4   2   0  ...    0.0   

        WP  HBP  BK    BFP    GF   R    SH   SF  GIDP  
35197  5.0  9.0   1  798.0   0.0  83  11.0  6.0  19.0  
10173  0.0  0.0   0  151.0   1.0  24   NaN  NaN   NaN  
35762  3.0  0.0   0  121.0   0.0  18   4.0  0.0   4.0  
14060  3.0  5.0   0  583.0  12.0  54   NaN  NaN   NaN  
16592  0.0  1.0   0   64.0   2.0  11   NaN  NaN   NaN  

[5 rows x 30 columns]

In [30]:
salaries.describe()

yearID        salary
count  26428.000000  2.642800e+04
mean    2000.878727  2.085634e+06
std        8.909314  3.455348e+06
min     1985.000000  0.000000e+00
25%     1994.000000  2.947020e+05
50%     2001.000000  5.500000e+05
75%     2009.000000  2.350000e+06
max     2016.000000  3.300000e+07

In [31]:
batting.describe()

/Users/Akihiro/.pyenv/versions/anaconda3-4.2.0/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


yearID          stint              G             AB  \
count  102816.000000  102816.000000  102816.000000  102816.000000   
mean     1964.262313       1.077838      51.343439     141.905511   
std        38.856297       0.284366      47.121658     184.654492   
min      1871.000000       1.000000       1.000000       0.000000   
25%      1934.000000       1.000000      13.000000       4.000000   
50%      1973.000000       1.000000      34.000000      49.000000   
75%      1998.000000       1.000000      80.000000     231.000000   
max      2016.000000       5.000000     165.000000     716.000000   

                   R              H             2B             3B  \
count  102816.000000  102816.000000  102816.000000  102816.000000   
mean       18.815544      37.139930       6.289167       1.293252   
std        28.242983      52.603757       9.662468       2.645770   
min         0.000000       0.000000       0.000000       0.000000   
25%         0.000000       0.000000       0.000000       0.000000   
50%         4.000000       9.000000       1.000000       0.000000   
75%        27.000000      58.000000       9.000000       1.000000   
max       192.000000     262.000000      67.000000      36.000000   

                  HR            RBI             SB            CS  \
count  102816.000000  102392.000000  101516.000000  79360.000000   
mean        2.813599      17.003975       2.976821      1.226008   
std         6.304919      26.352011       7.717174      2.747377   
min         0.000000       0.000000       0.000000      0.000000   
25%         0.000000            NaN            NaN           NaN   
50%         0.000000            NaN            NaN           NaN   
75%         2.000000            NaN            NaN           NaN   
max        73.000000     191.000000     138.000000     42.000000   

                  BB            SO           IBB            HBP            SH  \
count  102816.000000  94978.000000  66251.000000  100006.000000  96478.000000   
mean       13.067207     20.529712      1.105870       1.056057      2.299540   
std        20.746460     28.328542      2.780187       2.276251      4.241095   
min         0.000000      0.000000      0.000000       0.000000      0.000000   
25%         0.000000           NaN           NaN            NaN           NaN   
50%         3.000000           NaN           NaN            NaN           NaN   
75%        18.000000           NaN           NaN            NaN           NaN   
max       232.000000    223.000000    120.000000      51.000000     67.000000   

                 SF          GIDP  
count  66782.000000  76706.000000  
mean       1.054101      2.981018  
std        1.961732      4.735153  
min        0.000000      0.000000  
25%             NaN           NaN  
50%             NaN           NaN  
75%             NaN           NaN  
max       19.000000     36.000000

In [32]:
pitching.describe()

/Users/Akihiro/.pyenv/versions/anaconda3-4.2.0/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


yearID         stint             W             L             G  \
count  44963.000000  44963.000000  44963.000000  44963.000000  44963.000000   
mean    1968.670062      1.079643      4.715744      4.715722     23.681761   
std       37.569499      0.284834      5.813410      4.988846     18.493488   
min     1871.000000      1.000000      0.000000      0.000000      1.000000   
25%     1941.000000      1.000000      0.000000      1.000000      7.000000   
50%     1978.000000      1.000000      2.000000      3.000000     21.000000   
75%     2001.000000      1.000000      7.000000      8.000000     35.000000   
max     2016.000000      4.000000     59.000000     48.000000    106.000000   

                 GS            CG           SHO            SV        IPouts  \
count  44963.000000  44963.000000  44963.000000  44963.000000  44963.000000   
mean       9.485488      3.151035      0.448591      1.504793    253.872139   
std       12.283920      7.081844      1.109742      4.991518    257.330446   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      0.000000      0.000000      0.000000     49.000000   
50%        3.000000      0.000000      0.000000      0.000000    167.000000   
75%       17.000000      3.000000      0.000000      1.000000    393.000000   
max       75.000000     75.000000     16.000000     62.000000   2040.000000   

           ...                IBB            WP           HBP            BK  \
count      ...       30388.000000  44830.000000  44405.000000  44963.000000   
mean       ...           2.411379      2.528575      2.266096      0.301604   
std        ...           2.772648      3.425706      2.998881      0.755523   
min        ...           0.000000      0.000000      0.000000      0.000000   
25%        ...                NaN           NaN           NaN      0.000000   
50%        ...                NaN           NaN           NaN      0.000000   
75%        ...                NaN           NaN           NaN      0.000000   
max        ...          23.000000     63.000000     41.000000     16.000000   

                BFP            GF             R            SH            SF  \
count  44724.000000  44830.000000  44963.000000  12063.000000  12063.000000   
mean     343.312181      6.345260     43.021773      2.141922      1.878388   
std      348.758261      9.993949     43.245812      2.710191      2.121352   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%             NaN           NaN     11.000000           NaN           NaN   
50%             NaN           NaN     28.000000           NaN           NaN   
75%             NaN           NaN     67.000000           NaN           NaN   
max     2906.000000     84.000000    519.000000     21.000000     14.000000   

               GIDP  
count  12061.000000  
mean       5.322610  
std        5.990178  
min        0.000000  
25%             NaN  
50%             NaN  
75%             NaN  
max       40.000000  

[8 rows x 27 columns]

In [33]:
salaries[salaries.playerID == "happja01"]

yearID teamID lgID  playerID    salary
20357    2009    PHI   NL  happja01    405000
21188    2010    PHI   NL  happja01    470000
21761    2011    HOU   NL  happja01    474000
22587    2012    HOU   NL  happja01   2350000
23913    2013    TOR   AL  happja01   3700000
24710    2014    TOR   AL  happja01   5200000
25390    2015    SEA   AL  happja01   6700000
26386    2016    TOR   AL  happja01  10000000

In [34]:
batting[batting.playerID == "happja01"]

playerID  yearID  stint teamID lgID   G  AB  R  H  2B  ...   RBI   SB  \
89210   happja01    2007      1    PHI   NL   1   1  0  0   0  ...   0.0  0.0   
90603   happja01    2008      1    PHI   NL   8   7  0  0   0  ...   0.0  0.0   
91992   happja01    2009      1    PHI   NL  35  43  2  4   1  ...   1.0  0.0   
93347   happja01    2010      1    PHI   NL   3   4  0  0   0  ...   0.0  0.0   
93348   happja01    2010      2    HOU   NL  13  22  0  0   0  ...   0.0  0.0   
94726   happja01    2011      1    HOU   NL  29  43  3  7   1  ...   4.0  0.0   
96122   happja01    2012      1    HOU   NL  22  32  2  3   0  ...   1.0  0.0   
96123   happja01    2012      2    TOR   AL  10   0  0  0   0  ...   0.0  0.0   
97554   happja01    2013      1    TOR   AL  18   0  0  0   0  ...   0.0  0.0   
98962   happja01    2014      1    TOR   AL  30   4  0  0   0  ...   0.0  0.0   
100415  happja01    2015      1    SEA   AL  21   1  0  0   0  ...   0.0  0.0   
100416  happja01    2015      2    PIT   NL  11  22  1  2   0  ...   0.0  0.0   
101892  happja01    2016      1    TOR   AL  32   7  2  2   0  ...   0.0  0.0   

         CS  BB    SO  IBB  HBP   SH   SF  GIDP  
89210   0.0   0   1.0  0.0  0.0  0.0  0.0   0.0  
90603   0.0   0   4.0  0.0  0.0  2.0  0.0   0.0  
91992   0.0   1  15.0  0.0  0.0  7.0  0.0   0.0  
93347   0.0   1   2.0  0.0  0.0  1.0  0.0   0.0  
93348   0.0   0   7.0  0.0  0.0  2.0  0.0   0.0  
94726   0.0   3  17.0  0.0  0.0  7.0  0.0   0.0  
96122   0.0   3  16.0  0.0  0.0  5.0  0.0   0.0  
96123   0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  
97554   0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  
98962   0.0   0   4.0  0.0  0.0  0.0  0.0   0.0  
100415  0.0   1   0.0  0.0  0.0  0.0  0.0   1.0  
100416  0.0   0  13.0  0.0  0.0  3.0  0.0   0.0  
101892  0.0   0   3.0  0.0  0.0  3.0  0.0   0.0  

[13 rows x 22 columns]

### Try merge

In [35]:
data = salaries.merge(batting)

In [36]:
data[data.playerID == "happja01"]

yearID teamID lgID  playerID    salary  stint   G  AB  R  H  ...   RBI  \
19545    2009    PHI   NL  happja01    405000      1  35  43  2  4  ...   1.0   
20338    2010    PHI   NL  happja01    470000      1   3   4  0  0  ...   0.0   
20892    2011    HOU   NL  happja01    474000      1  29  43  3  7  ...   4.0   
21682    2012    HOU   NL  happja01   2350000      1  22  32  2  3  ...   1.0   
22978    2013    TOR   AL  happja01   3700000      1  18   0  0  0  ...   0.0   
23773    2014    TOR   AL  happja01   5200000      1  30   4  0  0  ...   0.0   
24450    2015    SEA   AL  happja01   6700000      1  21   1  0  0  ...   0.0   
25123    2016    TOR   AL  happja01  10000000      1  32   7  2  2  ...   0.0   

        SB   CS  BB    SO  IBB  HBP   SH   SF  GIDP  
19545  0.0  0.0   1  15.0  0.0  0.0  7.0  0.0   0.0  
20338  0.0  0.0   1   2.0  0.0  0.0  1.0  0.0   0.0  
20892  0.0  0.0   3  17.0  0.0  0.0  7.0  0.0   0.0  
21682  0.0  0.0   3  16.0  0.0  0.0  5.0  0.0   0.0  
22978  0.0  0.0   0   0.0  0.0  0.0  0.0  0.0   0.0  
23773  0.0  0.0   0   4.0  0.0  0.0  0.0  0.0   0.0  
24450  0.0  0.0   1   0.0  0.0  0.0  0.0  0.0   1.0  
25123  0.0  0.0   0   3.0  0.0  0.0  3.0  0.0   0.0  

[8 rows x 23 columns]

In [37]:
salaries.shape

(26428, 5)

In [38]:
batting.shape

(102816, 22)

In [39]:
data.shape

(25139, 23)

## About Datasets

>The updated version of the database contains complete batting and pitching statistics from 1871 to 2016, plus fielding statistics, standings, team stats, managerial records, post-season data, and more. For more details on the latest release, please read the documentation.

`from`
http://seanlahman.com/files/database/readme2016.txt

###  Batting Table

```
playerID       Player ID code
yearID         Year
stint          player's stint (order of appearances within a season)
teamID         Team
lgID           League
G              Games
AB             At Bats
R              Runs
H              Hits
2B             Doubles
3B             Triples
HR             Homeruns
RBI            Runs Batted In
SB             Stolen Bases
CS             Caught Stealing
BB             Base on Balls
SO             Strikeouts
IBB            Intentional walks
HBP            Hit by pitch
SH             Sacrifice hits
SF             Sacrifice flies
GIDP           Grounded into double plays
```

### Pitching Table

```
playerID       Player ID code
yearID         Year
stint          player's stint (order of appearances within a season)
teamID         Team
lgID           League
W              Wins
L              Losses
G              Games
GS             Games Started
CG             Complete Games 
SHO            Shutouts
SV             Saves
IPOuts         Outs Pitched (innings pitched x 3)
H              Hits
ER             Earned Runs
HR             Homeruns
BB             Walks
SO             Strikeouts
BAOpp          Opponent's Batting Average
ERA            Earned Run Average
IBB            Intentional Walks
WP             Wild Pitches
HBP            Batters Hit By Pitch
BK             Balks
BFP            Batters faced by Pitcher
GF             Games Finished
R              Runs Allowed
SH             Sacrifices by opposing batters
SF             Sacrifice flies by opposing batters
GIDP           Grounded into double plays by opposing batter
```


## Question

What are the characteristics of baseball players with the highest salaries?
- Hitter
- Pitcher 

## Data Cleaning

merging multiple files and handling missing values

In [40]:
## handling missing values
salaries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26428 entries, 0 to 26427
Data columns (total 5 columns):
yearID      26428 non-null int64
teamID      26428 non-null object
lgID        26428 non-null object
playerID    26428 non-null object
salary      26428 non-null int64
dtypes: int64(2), object(3)
memory usage: 1.0+ MB


In [41]:
batting.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102816 entries, 0 to 102815
Data columns (total 22 columns):
playerID    102816 non-null object
yearID      102816 non-null int64
stint       102816 non-null int64
teamID      102816 non-null object
lgID        102079 non-null object
G           102816 non-null int64
AB          102816 non-null int64
R           102816 non-null int64
H           102816 non-null int64
2B          102816 non-null int64
3B          102816 non-null int64
HR          102816 non-null int64
RBI         102392 non-null float64
SB          101516 non-null float64
CS          79360 non-null float64
BB          102816 non-null int64
SO          94978 non-null float64
IBB         66251 non-null float64
HBP         100006 non-null float64
SH          96478 non-null float64
SF          66782 non-null float64
GIDP        76706 non-null float64
dtypes: float64(9), int64(10), object(3)
memory usage: 17.3+ MB


The reason for deleting a row is that in the case of row deletion, about 30,000 pieces of data are lost.
CS,SO,SH,SF,GIDP 

In [42]:
drop_col = ["CS","SO","IBB","SH","SF","GIDP" ]
batting = batting.drop(drop_col, axis=1)

In [50]:
batting[batting.HBP > 1.0].tail(100)

playerID  yearID  stint teamID lgID    G   AB    R    H  2B  3B  HR  \
102348   penara02    2016      1    SFN   NL   30   87    9   26   6   1   1   
102351  peralda01    2016      1    ARI   NL   48  171   23   43   9   5   4   
102357  perazjo01    2016      1    CIN   NL   72  241   25   78   8   2   3   
102364  perezsa02    2016      1    KCA   AL  139  514   57  127  28   2  22   
102372   phamth01    2016      1    SLN   NL   78  159   26   36   7   0   9   
102373  phegljo01    2016      1    OAK   AL   26   78   11   20   6   0   1   
102375  phillbr01    2016      1    CIN   NL  141  550   74  160  34   1  11   
102377  pierzaj01    2016      1    ATL   NL   81  247   15   54  15   0   2   
102378  pillake01    2016      1    TOR   AL  146  548   59  146  35   2   7   
102385  piscost01    2016      1    SLN   NL  153  582   86  159  35   3  22   
102386  plaweke01    2016      1    NYN   NL   48  132    6   26   6   0   1   
102387  plouftr01    2016      1    MIN   AL   84  319   35   83  13   1  12   
102390  polanjo01    2016      1    MIN   AL   69  245   24   69  15   4   4   
102396  poseybu01    2016      1    SFN   NL  146  539   82  155  33   2  14   
102398  pradoma01    2016      1    MIA   NL  153  600   70  183  37   3   8   
102403  profaju01    2016      1    TEX   AL   90  272   35   65   6   3   5   
102404   puigya01    2016      1    LAN   NL  104  334   45   88  14   2  11   
102405  pujolal01    2016      1    LAA   AL  152  593   71  159  19   0  31   
102411  quinnro01    2016      1    PHI   NL   15   57   10   15   4   0   0   
102413  raburry01    2016      1    COL   NL  113  223   30   49  10   2   9   
102414  ramiral03    2016      1    SDN   NL  128  421   33  101  19   2   5   
102417  ramirha01    2016      1    BOS   AL  147  549   81  157  28   1  30   
102420  ramirjo01    2016      1    CLE   AL  152  565   84  176  46   3  11   
102429  ramoswi01    2016      1    WAS   NL  131  482   58  148  25   0  22   
102438  realmjt01    2016      1    MIA   NL  137  509   60  154  31   0  11   
102439  reckean01    2016      1    ATL   NL   33   90    6   25   8   0   2   
102449  rendoan01    2016      1    WAS   NL  156  567   91  153  38   2  20   
102451  reverbe01    2016      1    WAS   NL  103  350   44   76   9   7   2   
102455  reynoma01    2016      1    COL   NL  118  393   61  111  24   0  14   
102461  rickajo01    2016      1    BAL   AL   85  257   32   69  13   0   5   
...           ...     ...    ...    ...  ...  ...  ...  ...  ...  ..  ..  ..   
102655  suzukic01    2016      1    MIA   NL  143  327   48   95  15   5   1   
102656  suzukku01    2016      1    MIN   AL  106  345   34   89  24   1   8   
102670  teixema01    2016      1    NYA   AL  116  387   43   79  16   0  15   
102697  troutmi01    2016      1    LAA   AL  159  549  123  173  32   5  29   
102698  trumbma01    2016      1    BAL   AL  159  613   94  157  27   1  47   
102700  tuckepr01    2016      1    HOU   AL   48  134   11   22   8   1   4   
102703  tulowtr01    2016      1    TOR   AL  131  492   54  125  21   0  24   
102705  turneju01    2016      1    LAN   NL  151  556   79  153  34   3  27   
102708  uptonbj01    2016      1    SDN   NL   92  344   46   88  11   2  16   
102710  uptonju01    2016      1    DET   AL  153  570   81  140  28   2  31   
102713  uribeju01    2016      1    CLE   AL   73  238   19   49   9   0   7   
102714  utleych01    2016      1    LAN   NL  138  512   79  129  26   3  14   
102718  valenda01    2016      1    OAK   AL  130  471   72  135  22   1  17   
102719  vanslsc01    2016      1    LAN   NL   52  102   10   23   6   0   1   
102723  vazquch01    2016      1    BOS   AL   57  172   21   39   9   1   1   
102733  villajo01    2016      1    MIL   NL  156  589   92  168  38   3  19   
102738   vogtst01    2016      1    OAK   AL  137  490   54  123  30   2  14   
102740  vottojo01    2016      1    CIN   NL  158  556  101  181  34   2  29   
102747  w

In [22]:
batting = batting.dropna()


In [24]:
batting.shape

(66251, 17)

In [23]:
# drop pitch hittings
batting = batting[batting.HBP==false]

NameError: name 'false' is not defined

In [50]:
pitching.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44963 entries, 0 to 44962
Data columns (total 30 columns):
playerID    44963 non-null object
yearID      44963 non-null int64
stint       44963 non-null int64
teamID      44963 non-null object
lgID        44832 non-null object
W           44963 non-null int64
L           44963 non-null int64
G           44963 non-null int64
GS          44963 non-null int64
CG          44963 non-null int64
SHO         44963 non-null int64
SV          44963 non-null int64
IPouts      44963 non-null int64
H           44963 non-null int64
ER          44963 non-null int64
HR          44963 non-null int64
BB          44963 non-null int64
SO          44963 non-null int64
BAOpp       43437 non-null float64
ERA         44874 non-null float64
IBB         30388 non-null float64
WP          44830 non-null float64
HBP         44405 non-null float64
BK          44963 non-null int64
BFP         44724 non-null float64
GF          44830 non-null float64
R           4496

In [53]:
# drop 
batting = batting.dropna()
pitching = pitching.dropna()

In [55]:
batting.shape

(66175, 22)

In [56]:
pitching.shape

(12047, 30)

In [46]:
##  merging multiple files

In [58]:
pitchers_salaries = salaries.merge(pitching)
hitters_salaries = salaries.merge(batting)

In [61]:
print(pitchers_salaries.shape)
print(hitters_salaries.shape)

(6478, 31)
(25139, 23)


## Data Investigate

In [62]:
## corrlation
pitchers_salaries.corr()

yearID    salary     stint         W         L         G        GS  \
yearID  1.000000  0.147508  0.013190 -0.047809 -0.058990 -0.011041 -0.038154   
salary  0.147508  1.000000  0.025785  0.333042  0.281484 -0.071213  0.350308   
stint   0.013190  0.025785  1.000000 -0.023650 -0.018675 -0.039479 -0.011711   
W      -0.047809  0.333042 -0.023650  1.000000  0.673376  0.083832  0.842815   
L      -0.058990  0.281484 -0.018675  0.673376  1.000000  0.030210  0.829339   
G      -0.011041 -0.071213 -0.039479  0.083832  0.030210  1.000000 -0.237350   
GS     -0.038154  0.350308 -0.011711  0.842815  0.829339 -0.237350  1.000000   
CG     -0.094381  0.256018 -0.011524  0.584029  0.436783 -0.089794  0.562483   
SHO    -0.044715  0.211008 -0.009469  0.495354  0.339497 -0.074396  0.461552   
SV     -0.005123  0.084577 -0.010446 -0.117173 -0.090324  0.422989 -0.252132   
IPouts -0.059928  0.339936 -0.024809  0.910354  0.838396  0.090377  0.933087   
H      -0.089972  0.320402 -0.023379  0.857969  0.871140  0.023988  0.939500   
ER     -0.127606  0.285257 -0.022943  0.783026  0.887898 -0.031910  0.921041   
HR     -0.115037  0.283145 -0.021324  0.742481  0.811627 -0.036094  0.861382   
BB     -0.128184  0.210389 -0.030689  0.760397  0.781744  0.102040  0.821718   
SO      0.024816  0.355844 -0.022819  0.865726  0.721447  0.167739  0.823193   
BAOpp  -0.104634 -0.066531  0.023364 -0.161562 -0.010780 -0.341904 -0.005245   
ERA    -0.081903 -0.091818  0.009423 -0.215669 -0.112407 -0.336082 -0.092971   
IBB    -0.163588 -0.008637 -0.028061  0.248246  0.323291  0.388563  0.182354   
WP      0.012136  0.131597 -0.019058  0.480556  0.483829  0.135166  0.491180   
HBP    -0.092103  0.201272 -0.014050  0.594904  0.607077  0.057843  0.640136   
BK     -0.040902  0.061564 -0.004344  0.268993  0.259611  0.022903  0.288260   
BFP    -0.073577  0.330690 -0.025284  0.899664  0.854646  0.078371  0.939466   
GF     -0.029384 -0.043915 -0.017721 -0.239916 -0.222990  0.654938 -0.472049   
R      -0.129332  0.283778 -0.022410  0.784711  0.891949 -0.027252  0.922208   
SH     -0.127977  0.219743 -0.018431  0.615594  0.635485  0.089849  0.653136   
SF     -0.111708  0.193626 -0.020854  0.582895  0.642625  0.052153  0.657280   
GIDP   -0.074518  0.256511 -0.021983  0.764041  0.726141  0.064986  0.795770   

              CG       SHO        SV    ...          IBB        WP       HBP  \
yearID -0.094381 -0.044715 -0.005123    ...    -0.163588  0.012136 -0.092103   
salary  0.256018  0.211008  0.084577    ...    -0.008637  0.131597  0.201272   
stint  -0.011524 -0.009469 -0.010446    ...    -0.028061 -0.019058 -0.014050   
W       0.584029  0.495354 -0.117173    ...     0.248246  0.480556  0.594904   
L       0.436783  0.339497 -0.090324    ...     0.323291  0.483829  0.607077   
G      -0.089794 -0.074396  0.422989    ...     0.388563  0.135166  0.057843   
GS      0.562483  0.461552 -0.252132    ...     0.182354  0.491180  0.640136   
CG      1.000000  0.763406 -0.123093    ...     0.078671  0.253818  0.359840   
SHO     0.763406  1.000000 -0.101288    ...     0.077659  0.205671  0.289468   
SV     -0.123093 -0.101288  1.000000    ...     0.046517 -0.035888 -0.108810   
IPouts  0.598759  0.487162 -0.110046    ...     0.318277  0.547399  0.674335   
H       0.560948  0.442079 -0.161179    ...     0.319060  0.518737  0.663826   
ER      0.492291  0.378674 -0.194302    ...     0.298441  0.524769  0.669596   
HR      0.468158  0.355985 -0.166280    ...     0.232392  0.436178  0.595092   
BB      0.406696  0.337528 -0.120309    ...     0.379954  0.620323  0.657787   
SO      0.560609  0.473384 -0.016984    ...     0.267552  0.570299  0.610769   
BAOpp  -0.058299 -0.064172 -0.221503    ...    -0.072163 -0.108113 -0.069297   
ERA    -0.092498 -0.084532 -0.146026    ...    -0.148258 -0.127386 -0.119877   
IBB     0.078671  0.077659  0.046517    ...     1.000000  0.187635  0.244321   
WP      0.253818  0.205671 -0.035888    ...     0.187635  1.000000  0.433032   
HBP     0

In [63]:
hitters_salaries.corr()

yearID    salary     stint         G        AB         R         H  \
yearID  1.000000  0.350092 -0.042207  0.007237 -0.002148 -0.004633 -0.004427   
salary  0.350092  1.000000 -0.032819  0.176707  0.220555  0.241385  0.227602   
stint  -0.042207 -0.032819  1.000000 -0.070761 -0.048150 -0.044207 -0.046386   
G       0.007237  0.176707 -0.070761  1.000000  0.926647  0.891624  0.915721   
AB     -0.002148  0.220555 -0.048150  0.926647  1.000000  0.963351  0.989931   
R      -0.004633  0.241385 -0.044207  0.891624  0.963351  1.000000  0.973453   
H      -0.004427  0.227602 -0.046386  0.915721  0.989931  0.973453  1.000000   
2B      0.026711  0.231191 -0.045405  0.878513  0.948968  0.939122  0.958944   
3B     -0.035878  0.046366 -0.032159  0.615228  0.663957  0.673569  0.670524   
HR      0.028645  0.289477 -0.035121  0.740226  0.797799  0.848277  0.803997   
RBI    -0.001175  0.266974 -0.042269  0.864709  0.931072  0.939237  0.938857   
SB     -0.051576  0.064516 -0.024602  0.539113  0.591482  0.632885  0.600084   
CS     -0.101059  0.034538 -0.027668  0.617549  0.664289  0.672606  0.665094   
BB     -0.020472  0.255125 -0.041201  0.828553  0.876674  0.911328  0.875186   
SO      0.073886  0.226083 -0.045556  0.835671  0.899083  0.873130  0.866033   
IBB    -0.050190  0.239296 -0.028050  0.563258  0.590286  0.617353  0.610150   
HBP     0.084465  0.186924 -0.032892  0.612969  0.657541  0.666060  0.652923   
SH     -0.066124 -0.002474 -0.033081  0.196773  0.238143  0.182337  0.205830   
SF     -0.027379  0.197303 -0.034554  0.750413  0.805750  0.786370  0.806006   
GIDP    0.010042  0.231226 -0.041761  0.803890  0.868442  0.813446  0.864609   

              2B        3B        HR       RBI        SB        CS        BB  \
yearID  0.026711 -0.035878  0.028645 -0.001175 -0.051576 -0.101059 -0.020472   
salary  0.231191  0.046366  0.289477  0.266974  0.064516  0.034538  0.255125   
stint  -0.045405 -0.032159 -0.035121 -0.042269 -0.024602 -0.027668 -0.041201   
G       0.878513  0.615228  0.740226  0.864709  0.539113  0.617549  0.828553   
AB      0.948968  0.663957  0.797799  0.931072  0.591482  0.664289  0.876674   
R       0.939122  0.673569  0.848277  0.939237  0.632885  0.672606  0.911328   
H       0.958944  0.670524  0.803997  0.938857  0.600084  0.665094  0.875186   
2B      1.000000  0.604110  0.799898  0.924759  0.522837  0.591297  0.851195   
3B      0.604110  1.000000  0.411361  0.547427  0.657184  0.655564  0.544086   
HR      0.799898  0.411361  1.000000  0.927583  0.339380  0.398235  0.812870   
RBI     0.924759  0.547427  0.927583  1.000000  0.454501  0.527195  0.878883   
SB      0.522837  0.657184  0.339380  0.454501  1.000000  0.817372  0.516607   
CS      0.591297  0.655564  0.398235  0.527195  0.817372  1.000000  0.574464   
BB      0.851195  0.544086  0.812870  0.878883  0.516607  0.574464  1.000000   
SO      0.848414  0.562607  0.827672  0.874118  0.485302  0.555587  0.835937   
IBB     0.599679  0.320832  0.654648  0.666798  0.266722  0.310138  0.697307   
HBP     0.647502  0.398342  0.593202  0.640331  0.365883  0.411117  0.598251   
SH      0.152458  0.252117 -0.045916  0.075789  0.265021  0.292417  0.121494   
SF      0.793205  0.480992  0.703077  0.829636  0.402155  0.467585  0.734752   
GIDP    0.837937  0.466492  0.712633  0.841968  0.372466  0.466794  0.746473   

              SO       IBB       HBP        SH        SF      GIDP  
yearID  0.073886 -0.050190  0.084465 -0.066124 -0.027379  0.010042  
salary  0.226083  0.239296  0.186924 -0.002474  0.197303  0.231226  
stint  -0.045556 -0.028050 -0.032892 -0.033081 -0.034554 -0.041761  
G       0.835671  0.563258  0.612969  0.196773  0.750413  0.803890  
AB      0.899083  0.590286  0.657541  0.238143  0.805750  0.868442  
R       0.873130  0.617353  0.666060  0.182337  0.786370  0.813446  
H       0.866033  0.610150  0.652923  0.205830  0.806006  0.864609  
2B      0.848414  0.599679  0.647502  0.152458  0.793205  0.837937  
3B      0.562607  0.320832 